### PR TITLE
[CI] Support multiple windows versions

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

- Support Windows OS versions to be used in the CI as a list.
- Report the GH check as an overall rather than per tuple of module and windows version.
- Run more stages in parallel.

## Why is it important?

Keep the configuration simple.

### Cons

Number of concurrent agents: +44
BlueOcean visualisation does not show the parallel with parallel nicely!

image

### Pros

Easy to add more OS versions

## Issues
https://github.com/elastic/beats/pull/18382
https://github.com/elastic/beats/pull/18383
https://github.com/elastic/beats/pull/18384
https://github.com/elastic/beats/pull/18385
https://github.com/elastic/beats/pull/18386
https://github.com/elastic/beats/pull/18387
https://github.com/elastic/beats/pull/18388

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -691,19 +689,31 @@ def mageTarget(String context, String directory, String target) {
...
-def mageTargetWin(String context, String directory, String target) {
-  withGithubNotify(context: "${context}") {
-    withBeatsEnvWin() {
-      whenTrue(params.debug) {
-        dumpFilteredEnvironment()
-        dumpMageWin()
+def mageTargetWin(String context, String directory, String target, String label) {
+  return {
+    log(level: 'INFO', text: "context=${context} directory=${directory} target=${target} os=${label}")
+    node("immutable && ${label}"){
```
Comment (v1v) [inline Jenkinsfile:695]: See https://github.com/elastic/beats/pull/18388/commits/8161f6c9c784769f0d02b5eab3598f8647b5a300

```suggestion
    def immutable = label.equals('windows-7-32-bit') ? 'windows-immutable-32-bit' : 'windows-immutable'
    node("${immutable} && ${label}"){
```

File: Jenkinsfile [hunk 2]
```diff
@@ -9,6 +9,12 @@ import groovy.transform.Field
 */
 @Field def stashedTestReports = [:]
 
+
+/**
+ List of supported windows versions to be tested with
+*/
+@Field def windowsVersions = ['windows-2019', 'windows-2016', 'windows-2012-r2', 'windows-10', 'windows-2008', 'windows-2008-r2', 'windows-7', 'windows-7-32-bit']
```
Comment (v1v) [inline Jenkinsfile:16]:
```suggestion
@Field def windowsVersions = ['windows-2019', 'windows-2016', 'windows-2012-r2', 'windows-10', 'windows-2008-r2', 'windows-7', 'windows-7-32-bit']
```

**Comment by mikemadden42** [comment]
@v1v We'll need to update the support matrix [here](https://www.elastic.co/support/matrix) after this PR is merged.

**Comment by v1v** [comment]
This particular PR is not anymore required as we tackle the multi-windows support with:
- https://github.com/elastic/beats/pull/18941

The reason is a less complex CI build for the time being

**Pull request closed without merging**